### ISSUE TITLE
Batch processing of files

### ISSUE THREAD

**Issue by pchris7489** [mluis7/xml2xpath#14]
When using `xml2xpath.sh` to process a batch of many documents, it will stall

`cat my_document_list.csv| xargs -P 5 -{} -s "/" -a -l "{}"`

It should output each Xpath structure for each document in the list, but it will just stall.
This bug should be resolved by creating a seperate temporary file to process each document,

Additionnaly, having a command line parameter to write (tee) the output to a file (instead of `bash xml2xpath.sh -s "/" -a -l "{}" > my_output.txt`, use something like `bash xml2xpath.sh -s "/" -a -l "{}" -w "{}.txt"`) should help for batch processing also

**Comment by pchris7489** [comment]
I forgot to be more precise on this part "This bug should be resolved by creating a seperate temporary file to process each document,"
The process is eventually killed by other concurrent xml2xpath scripts, but also, the pipes seem to be using one same single file (the pipe file don't have a special name for each concurrent script).

**Comment by mluis7** [comment]
@pchris7489 
Yes, that could be another problem that will be fixed as (around line 430)
```
    fifo_in="xffin.$$"
    fifo_out="xffout.$$"
```

**Comment by mluis7** [comment]
Fixed by issue #15 .
Adding the ability to write the output to a file is out of scope for now. As a workaround, a simple wrapper script could help to get a log

```bash
#!/bin/bash
log_file="xml2xpath.$$.log"

./xml2xpath.sh [args + options] 2>&1 | tee "$log_file"
```
